android/app/src/main/java/app/organicmaps/sound/TtsPlayer.java

```diff
@@ -185,7 +191,31 @@ public void onError(String utteranceId, int errorCode) {
       mAudioFocusManager = new AudioFocusManager(context);
       mParams.putFloat(TextToSpeech.Engine.KEY_PARAM_VOLUME, Config.TTS.getVolume());
       mInitializing = false;
+      if (mReloadTriggered && sOnReloadCallback != null) {
```

Comment:
Are these variables really required? Can reinitialization + updateTts be called inside onChange() below?

nit: follow the code style.
```suggestion
      if (mReloadTriggered && sOnReloadCallback != null)
      {
```

Reply:
> Are these variables really required? 

I used `mReloadTriggered` to ensure the callback ( toast + `updateTts()`) is only used when there is system setting change(detected by `mTtsEngineObserver`)  , is it better to use the callback  at every initialize() call , including the first  ?


> Can reinitialization + updateTts be called inside onChange() below?

Do you mean calling `updateTts` within onChange()  instead of the callback in the fragment?
The `initialize()` sets up mTts in  asynchronous manner  right?,  Calling `updateTts` directly in `onChange` may lead to an update when tts engine has not loaded yet since its async , if im not wrong ? 
Also I'm not sure if passing the Fragment into `TtsPlayer.java` would be the best idea.